### ISSUE TITLE
Apache-AuthCookie's tests causes "can't call method ctx" 

### ISSUE THREAD

**Issue by kentfredric** [Test-More/test-more#757]
I haven't seen anything like this reported either on Test-More or Apache-AuthCookie

Affected:
```
Test-Simple-1.302075
Apache-AuthCookie-3.25
Apache-AuthCookie-3.26
```
Output
```
t/author-pod-syntax.t .. skipped: these tests are for testing by the author
t/signature.t .......... skipped: Set the environment variable TEST_SIGNATURE to enable this test.
t/util.t ............... ok                                             
t/real.t ............... 1/32 Can't call method "ctx" on an undefined value at /usr/lib64/perl5/vendor_perl/5.24.1/Test/Builder.pm line 228.
A context appears to have been destroyed without first calling release().
Based on $@ it does not look like an exception was thrown (this is not always
a reliable test)

This is a problem because the global error variables ($!, $@, and $?) will
not be restored. In addition some release callbacks will not work properly from
inside a DESTROY method.

Here are the context creation details, just in case a tool forgot to call
release():
  File: /usr/lib64/perl5/vendor_perl/5.24.1/Test/More.pm
  Line: 807
  Tool: Test::Builder::subtest

Cleaning up the CONTEXT stack...
A context appears to have been destroyed without first calling release().
Based on $@ it does not look like an exception was thrown (this is not always
a reliable test)

This is a problem because the global error variables ($!, $@, and $?) will
not be restored. In addition some release callbacks will not work properly from
inside a DESTROY method.

Here are the context creation details, just in case a tool forgot to call
release():
  File: t/real.t
  Line: 33
  Tool: Test::More::subtest

Cleaning up the CONTEXT stack...
t/real.t ............... Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 31/32 subtests 

Test Summary Report
-------------------
t/real.t             (Wstat: 65280 Tests: 1 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 32 tests but ran 1.
```

I don't see any code in the indicated test that would cause this.

Also note that the CPAN Testers output here is useless, because "pass"es there are actually "no tests ran" for some unhelpful reason.

http://www.cpantesters.org/cpan/report/90a2165a-e1b5-11e6-b58c-dcf18dff297d

**Comment by haarg** [comment]
Reduced:
```
use strict;
use warnings;

use Test::More;

subtest 'subtest' => sub {
    Test::Builder->new->reset;

    ok 1;
};

done_testing;
```

In the `t/real.t` test, it is calling `plan` in the subtest.  Apache::Test provides its own plan function which includes a `$tb->reset` call: https://metacpan.org/source/SHAY/Apache-Test-1.40/lib/Apache/Test.pm#L192

**Comment by haarg** [comment]
Alternatively:
```
use strict;
use warnings;

use Test::More;
use Apache::Test -withtestmore;

subtest 'subtest' => sub {
    plan tests => 1;

    ok 1;
};

done_testing;
```
for the failure while using Apache::Test

**Comment by exodist** [comment]
Also tested with Apache::Test, appears fixed.

**Comment by exodist** [comment]
Uploaded Test-Simple-1.302076-TRIAL.tar.gz  withthe fix, assuming no issues I will release as stable in a couple days.

**Comment by kentfredric** [comment]
Joys. Guess which version of Test::Simple shipped with Perl 5.26 

```
 corelist -a Test::More  | grep 5.26
  v5.26.0    1.302073  
```
